src/dev/impl/DevToys/ViewModels/Tools/Graphic/ImageConverter/ImageConversionWorkItem.cs

```diff
@@ -0,0 +1,133 @@
+
+namespace DevToys.ViewModels.Tools.ImageConverter
+{
+    internal sealed class ImageConversionWorkItem : ObservableRecipient
+    {
+        private string _fileSize = string.Empty;
+        private readonly Action<Exception> _errorHandler = HandleError;
+
+        internal ImageConverterStrings Strings => LanguageManager.Instance.ImageConverter;
+
+        internal string FileName { get; }
+
+        internal string FilePath { get; }
+
+        internal SoftwareBitmap? Bitmap { get; private set; }
+
+        internal string FileSize
+        {
+            get => _fileSize;
+            private set
+            {
+                ThreadHelper.ThrowIfNotOnUIThread();
+                SetProperty(ref _fileSize, value);
+            }
+        }
+
+        internal string? ErrorMessage { get; private set; }
+
+        internal event EventHandler? DeleteItemRequested;
+
+        internal ImageConversionWorkItem(StorageFile file)
+        {
+            Arguments.NotNull(file, nameof(file));
+
+            FileName = file.Name;
+            FilePath = file.Path;
+
+            DeleteCommand = new RelayCommand(ExecuteDeleteCommand);
+            SaveCommand = new AsyncRelayCommand<ImageConverterToolViewModel>(ExecuteSaveCommandAsync);
+
+            ComputePropertiesAsync(file).ForgetSafely(_errorHandler);
+            DecodeImageAsync(file).ForgetSafely(_errorHandler);
+        }
+
+        #region DeleteCommand
+
+        public IRelayCommand DeleteCommand { get; }
+
+        private void ExecuteDeleteCommand()
+        {
+            DeleteItemRequested?.Invoke(this, EventArgs.Empty);
+        }
+
+        #endregion
+
+        #region SaveCommand
+
+        public IAsyncRelayCommand SaveCommand { get; }
+
+        private async Task ExecuteSaveCommandAsync(ImageConverterToolViewModel viewModel)
+        {
+            string? fileExtension = ImageHelper.GetExtension(viewModel.ConvertedFormat);
+
+            var savePicker = new FileSavePicker
+            {
+                SuggestedStartLocation = PickerLocationId.ComputerFolder
+            };
+
+            savePicker.FileTypeChoices.Add(
+                fileExtension.Replace(".", string.Empty).ToUpperInvariant(),
+                new List<string>() { fileExtension!.ToLowerInvariant() });
+
+            StorageFile? newFile = await savePicker.PickSaveFileAsync();
+
+            if (newFile is not null)
+            {
+                using (IRandomAccessStream outputStream = await newFile.OpenAsync(FileAccessMode.ReadWrite))
+                {
+                    BitmapEncoder? encoder = await ImageHelper.GetEncoderAsync(viewModel.ConvertedFormat, outputStream);
+
+                    encoder.SetSoftwareBitmap(Bitmap);
+                    await encoder.FlushAsync();
+                }
+
+                DeleteCommand.Execute(this);
+            }
+        }
+
+        #endregion
+
+        private static void HandleError(Exception e)
+        {
+            throw e;
```

Comment:
I'm not following, where will this be catch?
Perhaps we should consider using `Logger.LogFaultAsync` instead.

Reply:
The purpose here is to raise an exception that wouldn´t be raised because it is being executed in a background (worker) thread, so it is possible to catch it in the ViewModel (ImageConversionWorkItem constructor is under a try). If you don´t think this approach will work, should we raise the exception in HandleError method using the UI thread? 

```
private async static void HandleError(Exception e)
{
      await ThreadHelper.RunOnUIThreadAsync(() => { throw e; });
}
```
or should we just log here the error?

Reply:
> The purpose here is to raise an exception that wouldn´t be raised because it is being executed in a background (worker) thread

It will still be raised, just not caught by anything. I'd suggest to just log the error in the Logger. :)

Reply:
> It will still be raised

Cool, so anyway error will be logged then in ViewModel